headless-server-commerce-base-lib/src/main/java/com/coremedia/blueprint/headlessserver/CommerceCaasConfig.java

```diff
@@ -0,0 +1,410 @@
+import org.springframework.cache.caffeine.CaffeineCache;
+import org.springframework.cache.support.SimpleCacheManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.expression.MapAccessor;
+import org.springframework.context.support.ConversionServiceFactoryBean;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.expression.BeanResolver;
+import org.springframework.expression.PropertyAccessor;
+import org.springframework.expression.spel.support.ReflectivePropertyAccessor;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.filter.CommonsRequestLoggingFilter;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.PathMatchConfigurer;
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.coremedia.caas.headless_server.plugin_support.PluginSupport.QUALIFIER_CAAS_FILTER_PREDICATE;
+import static com.coremedia.caas.headless_server.plugin_support.PluginSupport.QUALIFIER_CAAS_WIRING_FACTORY;
+import static com.coremedia.caas.headless_server.plugin_support.PluginSupport.QUALIFIER_PLUGIN_FILTER_PREDICATE;
+import static com.coremedia.caas.headless_server.plugin_support.PluginSupport.QUALIFIER_PLUGIN_WIRING_FACTORIES;
+import static java.lang.invoke.MethodHandles.lookup;
+import static java.util.Collections.emptyList;
+
+@Configuration(proxyBeanMethods = false)
+@EnableConfigurationProperties({
+        CaasServiceConfigurationProperties.class,
+        CaasGraphqlConfigurationProperties.class,
+        RemoteServiceConfigurationProperties.class,
+        StaxContextConfigurationProperties.class,
+        ContentConfigurationProperties.class,
+        DoctypeConfigurationProperties.class,
+})
+@EnableWebMvc
+@Import({
+        // inspired by CaasConfig
+        CacheConfiguration.class,
+        CustomizerConfiguration.class,
+        CapRepositoriesConfiguration.class,
+        GraphQlConfiguration.class,
+        PluginConfiguration.class,
+        CaasWebConfig.class,
+        GraphQLErrorController.class,
+})
+
+/**
+ * Copy and adapt the configuration from the CaasConfig class.
+ */
+public class CommerceCaasConfig implements WebMvcConfigurer {
```

Comment:
Most of this stuff seems to be redundant with CaasConfig. Create a jira issue to extract a public API configuration class from CaasConfig in some core or bpbase module that can be imported by CaasConfig and CommerceCaasConfig to avoid all the redundant beans.

Reply:
https://coremediagmbh.atlassian.net/browse/CMS-24792